include/sys/zfs_sa.h

```diff
@@ -87,6 +88,8 @@ typedef enum zpl_attr {
 #define	SA_UID_OFFSET		24
 #define	SA_GID_OFFSET		32
 #define	SA_PARENT_OFFSET	40
+#define	SA_FLAGS_OFFSET		48
+#define	SA_PROJID_OFFSET	128
```

Comment:
I have already put the PROJID just after the LINKS and before the RDEV. The new offset is really 128 as following:

MODE 0
SIZE 8
GEN 16
UID 24
GID 32
PARENT 40
FALGS 48
ATIME 56
MTIME 72
CTIME 88
CRTIME 104
LINKS 120
PROJID 128